engine/src/physics/solver/penetration_constraint/plugin.cpp

```diff
@@ -111,6 +101,68 @@ void solvePenetrationConstraint(Query<Entity, const Mass&, AccumulatedCorrection
             }
         }
 
+        // Friction
+        // for each contact point in the future, since we have particles now we only need to do it once
+        {
+            float normalImpulse = constraint.normalImpulse;
+            float frictionImpulse1 = constraint.frictionImpulse1;
+            float frictionImpulse2 = constraint.frictionImpulse2;
+
+            // Relative velocity at contact
+            glm::vec3 vr2;
+            glm::vec3 vr1;
+            if (ent1 == constraint.entity)
+            {
+                vr2 = velocity2.vec;
+                vr1 = velocity1.vec;
+            }
+            else
+            {
+                vr2 = velocity1.vec;
+                vr1 = velocity2.vec;
+            }
+
+            glm::vec3 vr = vr2 - vr1;
+            glm::vec3 tangent1 = vr - glm::dot(vr, constraint.normal) * constraint.normal;
+            glm::vec3 tangent2;
+            float tangentLenSq = glm::length2(tangent1);
+            if (tangentLenSq > 1e-6)
```

Comment:
Maybe we should also use a constant here? We might have some problems regarding floating point precision.
@luishfonseca what do you think?

Reply:
1e-6 is huge, not really sure what is the impact of erroneously skipping this block

Reply:
I did leave the wrong number here, the place where I took this from suggests "epsilon * epsilon", they didn't say what epsilon should be but if I recall I saw that bullet had it at 1e-6 back in like 2007 so that's what I used and forgot to multiply. The impact of skipping would be that you wouldn't have friction at that step, from what I read it's better to have a lower epsilon the faster the objects can move, essencially we're just making friction more sensitive to tiny movements, gonna try and find where I read that from.